src/components/MailberryForm.tsx

```diff
@@ -1,155 +1,192 @@
-        <div className="MBerror-wrapper">
-          <p className="MBerror-message">
-            Something went wrong.
-          </p>
-        </div>
-      )}
-      </form>
-    </div>
-  );
-};
-
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { FieldType, FormPopupOptions, FormatOptions, MailberryFormCSSProps, formFormat } from '../types';
+import { css } from '../utils/css-generator';
+import MailberryFormFieldComponents from "./MailberryFormField";
+import MailberryFormPopup from './MailberryPopupOption';
+import MailberryFormSnippet from './MailberrySnippetOption';
+
+type ContextProps = {
+  fields: FieldType[];
+  setFields: React.Dispatch<React.SetStateAction<FieldType[]>>;
+  emptyFields: boolean;
+  invalidEmail: boolean;
+  isSubmitting: boolean;
+  setIsSubmitting: React.Dispatch<React.SetStateAction<boolean>>;
+  isSubmitted: boolean;
+  setIsSubmitted: React.Dispatch<React.SetStateAction<boolean>>;
+  showErrorMessage: boolean;
+  showThanksMessage: boolean;
+}
+
+export const FormContext = createContext({} as ContextProps);
+
+const API_FORM_URL = 'https://backend.mailberry.ai/public'
+
+type MailberryFormProps = {
+  formId: string;
+  formStyle?: MailberryFormCSSProps;
+  signature?: boolean;
+  thanksMessage: string;
+  format: FormatOptions;
+  showAt?: FormPopupOptions;
+  children: React.ReactNode | React.ReactNode[];
+};
+
+interface MailberryFormComponents {
+  EmailInput: typeof MailberryFormFieldComponents.MailberryEmailInput;
+  TextInput: typeof MailberryFormFieldComponents.MailberryTextInput;
+  NumberInput: typeof MailberryFormFieldComponents.MailberryNumberInput;
+  DateInput: typeof MailberryFormFieldComponents.MailberryDateInput;
+  Description: typeof MailberryDescription;
+  FieldError: typeof MailberryFieldError;
+  Submit: typeof MailberrySubmit;
+  ThanksMessage: typeof MailberryThanksMessage;
+}
+
+const MailberryForm: React.FC<MailberryFormProps> & MailberryFormComponents = ({ formId, formStyle, signature = true, thanksMessage, format, showAt = 'IMMEDIATELY', children }): React.ReactNode => {
+  const href = `${API_FORM_URL}/${formId}`;
+
+  const [fields, setFields] = useState<FieldType[]>([]);
+  const [emptyFields, setEmptyFields] = useState(false);
+  const [invalidEmail, setInvalidEmail] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [showThanksMessage, setShowThanksMessage] = useState(false);
+  const [showErrorMessage, setShowErrorMessage] = useState(false);
+
+  useEffect(() => {
+    if(!formStyle) return
+
+    if(document.getElementById('mailberry-form-styles')) return
+    console.log({formStyle})
+
+    var styleTag = document.createElement('style');
+    styleTag.innerHTML = css(JSON.parse(JSON.stringify(formStyle)));
+    styleTag.setAttribute('id', 'mailberry-form-styles');
+    document.getElementsByTagName('head')[0].appendChild(styleTag);

```

Comment:
avoid using a <style> tag for this lib, just remove it and add the styles inline to each input